internal/repository/repo.go

```diff
@@ -0,0 +1,117 @@
+package repository
+
+import (
+	model "github.com/cyneruxyz/address-book/gen/proto"
+	"hash/fnv"
+	"strings"
+	"sync"
+)
+
+type AddressBook struct {
+	sync.Mutex
+	Names  map[string][]*model.AddressField
+	Addrs  map[string][]*model.AddressField
+	Phones map[string][]*model.AddressField
+	Hashes map[uint32]*model.AddressField
```

Comment:
1. There is no needed to use generated structure with extra fields; it's better to create new structure for storing the data in memory.
2. Could you please explain, why to use 4 separate maps? It might be used, but the task does not have any requirements to be satisfied with adding extra maps. Also, we will have extra memory consumption in the implemented approach.
3. It's not a request to re-implement the storage yet. But please, explain the solution, especially how the wildcards can be applied in Find operation?